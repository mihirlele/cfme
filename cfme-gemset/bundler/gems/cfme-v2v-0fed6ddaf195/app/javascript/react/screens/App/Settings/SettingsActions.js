import { reset } from 'redux-form';
import URI from 'urijs';
import API from '../../../../common/API';

import {
  V2V_FETCH_SERVERS,
  V2V_FETCH_SETTINGS,
  V2V_PATCH_SETTINGS,
  FETCH_V2V_CONVERSION_HOSTS,
  FETCH_V2V_CONVERSION_HOST_TASKS,
  SHOW_V2V_CONVERSION_HOST_WIZARD,
  HIDE_V2V_CONVERSION_HOST_WIZARD,
  V2V_CONVERSION_HOST_WIZARD_EXITED,
  POST_V2V_CONVERSION_HOSTS,
  SET_V2V_CONVERSION_HOST_TO_DELETE,
  SHOW_V2V_CONVERSION_HOST_DELETE_MODAL,
  HIDE_V2V_CONVERSION_HOST_DELETE_MODAL,
  DELETE_V2V_CONVERSION_HOST,
  SET_V2V_CONVERSION_HOST_TASK_TO_RETRY,
  SHOW_V2V_CONVERSION_HOST_RETRY_MODAL,
  HIDE_V2V_CONVERSION_HOST_RETRY_MODAL,
  V2V_CONVERSION_HOST_RETRY_MODAL_EXITED
} from './SettingsConstants';
import { getApiSettingsFromFormValues } from './helpers';
import { stepIDs } from './screens/ConversionHostsSettings/components/ConversionHostWizard/ConversionHostWizardConstants';

const _getServersActionCreator = url => dispatch =>
  dispatch({
    type: V2V_FETCH_SERVERS,
    payload: API.get(url)
  });

export const fetchServersAction = url => {
  const uri = new URI(url);
  return _getServersActionCreator(uri.toString());
};

const _getSettingsActionCreator = url => dispatch =>
  dispatch({
    type: V2V_FETCH_SETTINGS,
    payload: API.get(url)
  });

export const fetchSettingsAction = url => {
  const uri = new URI(url);
  return _getSettingsActionCreator(uri.toString());
};

const _patchSettingsActionCreator = (urls, settingsFormValues) => dispatch => {
  const patchBody = getApiSettingsFromFormValues(settingsFormValues);
  return dispatch({
    type: V2V_PATCH_SETTINGS,
    payload: new Promise((resolve, reject) => {
      Promise.all(urls.map(url => API.patch(url, patchBody)))
        .then(payloads => resolve(payloads[0]))
        .catch(e => reject(e));
    })
  });
};

export const patchSettingsAction = (servers, newSettings) => {
  const settingsUrls = servers.map(server => new URI(`${server.href}/settings`).toString());
  return _patchSettingsActionCreator(settingsUrls, newSettings);
};

const _getConversionHostsActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_CONVERSION_HOSTS,
    payload: API.get(url)
  });

export const fetchConversionHostsAction = url => {
  const uri = new URI(url);
  return _getConversionHostsActionCreator(uri.toString());
};

const _getConversionHostTasksActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_CONVERSION_HOST_TASKS,
    payload: API.get(url)
  });

export const fetchConversionHostTasksAction = url => {
  const uri = new URI(url);
  return _getConversionHostTasksActionCreator(uri.toString());
};

export const showConversionHostWizardAction = () => dispatch => dispatch({ type: SHOW_V2V_CONVERSION_HOST_WIZARD });

export const hideConversionHostWizardAction = () => dispatch => dispatch({ type: HIDE_V2V_CONVERSION_HOST_WIZARD });

export const conversionHostWizardExitedAction = () => dispatch => {
  dispatch({ type: V2V_CONVERSION_HOST_WIZARD_EXITED });
  // Dispatch reset for all the wizard step forms here
  Object.values(stepIDs).forEach(formName => {
    dispatch(reset(formName));
  });
};

const _postConversionHostsActionCreator = (url, postBodies) => dispatch =>
  dispatch({
    type: POST_V2V_CONVERSION_HOSTS,
    payload: Promise.all(postBodies.map(body => API.post(url, body)))
  });

export const postConversionHostsAction = (url, postBodies) =>
  _postConversionHostsActionCreator(new URI(url).toString(), postBodies);

export const setHostToDeleteAction = host => ({
  type: SET_V2V_CONVERSION_HOST_TO_DELETE,
  payload: host
});

export const showConversionHostDeleteModalAction = () => ({
  type: SHOW_V2V_CONVERSION_HOST_DELETE_MODAL
});

export const hideConversionHostDeleteModalAction = () => ({
  type: HIDE_V2V_CONVERSION_HOST_DELETE_MODAL
});

export const _deleteConversionHostActionCreator = (url, host) => dispatch =>
  dispatch({
    type: DELETE_V2V_CONVERSION_HOST,
    payload: new Promise((resolve, reject) => {
      API.post(`${url}/${host.id}`, {
        action: 'delete'
      })
        .then(response => {
          resolve(response);
        })
        .catch(e => reject(e));
    })
  });

export const deleteConversionHostAction = (url, host) =>
  _deleteConversionHostActionCreator(new URI(url).toString(), host);

export const setConversionHostTaskToRetryAction = task => ({
  type: SET_V2V_CONVERSION_HOST_TASK_TO_RETRY,
  payload: task
});

export const showConversionHostRetryModalAction = () => ({
  type: SHOW_V2V_CONVERSION_HOST_RETRY_MODAL
});

export const hideConversionHostRetryModalAction = () => ({
  type: HIDE_V2V_CONVERSION_HOST_RETRY_MODAL
});

export const conversionHostRetryModalExitedAction = () => ({
  type: V2V_CONVERSION_HOST_RETRY_MODAL_EXITED
});
