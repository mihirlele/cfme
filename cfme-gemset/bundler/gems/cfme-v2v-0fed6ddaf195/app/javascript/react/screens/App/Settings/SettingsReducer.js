import Immutable from 'seamless-immutable';

import {
  V2V_FETCH_SERVERS,
  V2V_FETCH_SETTINGS,
  V2V_PATCH_SETTINGS,
  FETCH_V2V_CONVERSION_HOSTS,
  FETCH_V2V_CONVERSION_HOST_TASKS,
  SHOW_V2V_CONVERSION_HOST_WIZARD,
  HIDE_V2V_CONVERSION_HOST_WIZARD,
  V2V_CONVERSION_HOST_WIZARD_EXITED,
  POST_V2V_CONVERSION_HOSTS,
  SET_V2V_CONVERSION_HOST_TO_DELETE,
  SHOW_V2V_CONVERSION_HOST_DELETE_MODAL,
  HIDE_V2V_CONVERSION_HOST_DELETE_MODAL,
  DELETE_V2V_CONVERSION_HOST,
  SET_V2V_CONVERSION_HOST_TASK_TO_RETRY,
  SHOW_V2V_CONVERSION_HOST_RETRY_MODAL,
  HIDE_V2V_CONVERSION_HOST_RETRY_MODAL,
  V2V_CONVERSION_HOST_RETRY_MODAL_EXITED
} from './SettingsConstants';

import {
  getFormValuesFromApiSettings,
  parseConversionHostTasksMetadata,
  indexConversionHostTasksByResource
} from './helpers';

export const initialState = Immutable({
  conversionHosts: [],
  conversionHostTasks: [],
  conversionHostTasksByResource: {},
  conversionHostToDelete: null,
  conversionHostDeleteModalVisible: false,
  conversionHostRetryModalMounted: false,
  conversionHostRetryModalVisible: false,
  conversionHostTaskToRetry: null,
  conversionHostWizardMounted: false,
  conversionHostWizardVisible: false,
  errorDeleteConversionHost: false,
  errorFetchingConversionHosts: null,
  errorFetchingConversionHostTasks: null,
  errorFetchingServers: null,
  errorFetchingSettings: null,
  errorPostingConversionHosts: null,
  errorSavingSettings: null,
  fetchingServersRejected: false,
  fetchingSettingsRejected: false,
  isDeletingConversionHost: false,
  isFetchingConversionHosts: false,
  isFetchingConversionHostTasks: false,
  isFetchingServers: false,
  isFetchingSettings: false,
  isPostingConversionHosts: false,
  isRejectedDeletingConversionHost: false,
  isRejectedFetchingConversionHosts: false,
  isRejectedFetchingConversionHostTasks: false,
  isRejectedPostingConversionHosts: false,
  isSavingSettings: false,
  postConversionHostsResults: [],
  savedSettings: {},
  savingSettingsRejected: false,
  servers: []
});

export default (state = initialState, action) => {
  switch (action.type) {
    case `${V2V_FETCH_SERVERS}_PENDING`:
      return state
        .set('isFetchingServers', true)
        .set('fetchingServersRejected', false)
        .set('errorFetchingServers', null);
    case `${V2V_FETCH_SERVERS}_REJECTED`:
      return state
        .set('isFetchingServers', false)
        .set('fetchingServersRejected', true)
        .set('errorFetchingServers', action.payload);
    case `${V2V_FETCH_SERVERS}_FULFILLED`:
      return state
        .set('isFetchingServers', false)
        .set('fetchingServersRejected', false)
        .set('errorFetchingServers', null)
        .set('servers', action.payload.data.resources);

    case `${V2V_FETCH_SETTINGS}_PENDING`:
      return state
        .set('isFetchingSettings', true)
        .set('fetchingSettingsRejected', false)
        .set('errorFetchingSettings', null);
    case `${V2V_FETCH_SETTINGS}_REJECTED`:
      return state
        .set('isFetchingSettings', false)
        .set('fetchingSettingsRejected', true)
        .set('errorFetchingSettings', action.payload);
    case `${V2V_FETCH_SETTINGS}_FULFILLED`:
      return state
        .set('isFetchingSettings', false)
        .set('fetchingSettingsRejected', false)
        .set('errorFetchingSettings', null)
        .set('savedSettings', getFormValuesFromApiSettings(action.payload.data));

    case `${V2V_PATCH_SETTINGS}_PENDING`:
      return state
        .set('isSavingSettings', true)
        .set('savingSettingsRejected', false)
        .set('errorSavingSettings', null);
    case `${V2V_PATCH_SETTINGS}_REJECTED`:
      return state
        .set('isSavingSettings', false)
        .set('savingSettingsRejected', true)
        .set('errorSavingSettings', action.payload);
    case `${V2V_PATCH_SETTINGS}_FULFILLED`:
      return state
        .set('isSavingSettings', false)
        .set('savingSettingsRejected', false)
        .set('errorSavingSettings', null)
        .set('savedSettings', getFormValuesFromApiSettings(action.payload));

    case `${FETCH_V2V_CONVERSION_HOSTS}_PENDING`:
      return state
        .set('isFetchingConversionHosts', true)
        .set('isRejectedFetchingConversionHosts', false)
        .set('errorFetchingConversionHosts', null);
    case `${FETCH_V2V_CONVERSION_HOSTS}_FULFILLED`:
      return state
        .set('conversionHosts', action.payload.data.resources)
        .set('isFetchingConversionHosts', false)
        .set('isRejectedFetchingConversionHosts', false)
        .set('errorFetchingConversionHosts', null);
    case `${FETCH_V2V_CONVERSION_HOSTS}_REJECTED`:
      return state
        .set('isFetchingConversionHosts', false)
        .set('isRejectedFetchingConversionHosts', true)
        .set('errorFetchingConversionHosts', action.payload);

    case `${FETCH_V2V_CONVERSION_HOST_TASKS}_PENDING`:
      return state
        .set('isFetchingConversionHostTasks', true)
        .set('isRejectedFetchingConversionHostTasks', false)
        .set('errorFetchingConversionHostTasks', null);
    case `${FETCH_V2V_CONVERSION_HOST_TASKS}_FULFILLED`: {
      const tasksWithMetadata = parseConversionHostTasksMetadata(action.payload.data.resources);
      const tasksByResource = indexConversionHostTasksByResource(tasksWithMetadata);
      return state
        .set('conversionHostTasks', tasksWithMetadata)
        .set('conversionHostTasksByResource', tasksByResource)
        .set('isFetchingConversionHostTasks', false)
        .set('isRejectedFetchingConversionHostTasks', false)
        .set('errorFetchingConversionHostTasks', null);
    }
    case `${FETCH_V2V_CONVERSION_HOST_TASKS}_REJECTED`:
      return state
        .set('isFetchingConversionHostTasks', false)
        .set('isRejectedFetchingConversionHostTasks', true)
        .set('errorFetchingConversionHostTasks', action.payload);

    case SHOW_V2V_CONVERSION_HOST_WIZARD:
      return state.set('conversionHostWizardMounted', true).set('conversionHostWizardVisible', true);
    case HIDE_V2V_CONVERSION_HOST_WIZARD:
      return state.set('conversionHostWizardVisible', false);
    case V2V_CONVERSION_HOST_WIZARD_EXITED:
      return state.set('conversionHostWizardMounted', false);

    case `${POST_V2V_CONVERSION_HOSTS}_PENDING`:
      return state
        .set('isPostingConversionHosts', true)
        .set('isRejectedPostingConversionHosts', false)
        .set('errorPostingConversionHosts', false);
    case `${POST_V2V_CONVERSION_HOSTS}_FULFILLED`:
      return state
        .set('isPostingConversionHosts', false)
        .set('isRejectedPostingConversionHosts', false)
        .set('errorPostingConversionHosts', null)
        .set('postConversionHostsResults', action.payload);
    case `${POST_V2V_CONVERSION_HOSTS}_REJECTED`:
      return state
        .set('isPostingConversionHosts', false)
        .set('isRejectedPostingConversionHosts', true)
        .set('errorPostingConversionHosts', action.payload);
    case SET_V2V_CONVERSION_HOST_TO_DELETE:
      return state.set('conversionHostToDelete', action.payload);
    case SHOW_V2V_CONVERSION_HOST_DELETE_MODAL:
      return state.set('conversionHostDeleteModalVisible', true);
    case HIDE_V2V_CONVERSION_HOST_DELETE_MODAL:
      return state.set('conversionHostDeleteModalVisible', false);

    case `${DELETE_V2V_CONVERSION_HOST}_PENDING`:
      return state.set('isDeletingConversionHost', action.payload);
    case `${DELETE_V2V_CONVERSION_HOST}_FULFILLED`:
      return state
        .set('deleteConversionHostResponse', action.payload.data)
        .set('isDeletingConversionHost', null)
        .set('isRejectedDeletingConversionHost', false)
        .set('errorDeleteConversionHost', null)
        .set('conversionHostDeleteModalVisible', false);
    case `${DELETE_V2V_CONVERSION_HOST}_REJECTED`:
      return state
        .set('errorDeleteConversionHost', action.payload)
        .set('isRejectedDeletingConversionHost', true)
        .set('isDeletingConversionHost', null)
        .set('conversionHostDeleteModalVisible', false);

    case SET_V2V_CONVERSION_HOST_TASK_TO_RETRY:
      return state.set('conversionHostTaskToRetry', action.payload);
    case SHOW_V2V_CONVERSION_HOST_RETRY_MODAL:
      return state.set('conversionHostRetryModalMounted', true).set('conversionHostRetryModalVisible', true);
    case HIDE_V2V_CONVERSION_HOST_RETRY_MODAL:
      return state.set('conversionHostRetryModalVisible', false);
    case V2V_CONVERSION_HOST_RETRY_MODAL_EXITED:
      return state.set('conversionHostRetryModalMounted', false);

    default:
      return state;
  }
};
