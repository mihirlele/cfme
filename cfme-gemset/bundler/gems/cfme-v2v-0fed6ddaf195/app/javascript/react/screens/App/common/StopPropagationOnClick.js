import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

// This component has no role on the page, and will be hidden from screen readers (role="presentation").
// Prevents a click on an inner element (e.g. a DropdownKebab) from triggering a click on an ancestor (e.g. a table row).
// The onKeyUp handler and role are used to satisfy eslint-plugin-jsx-a11y.
const StopPropagationOnClick = ({ children }) => (
  <span onClick={e => e.stopPropagation()} onKeyUp={noop} role="presentation">
    {children}
  </span>
);

StopPropagationOnClick.propTypes = {
  children: PropTypes.node
};

export default StopPropagationOnClick;
