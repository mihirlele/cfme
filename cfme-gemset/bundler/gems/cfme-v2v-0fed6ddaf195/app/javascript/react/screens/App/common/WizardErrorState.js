import React from 'react';
import PropTypes from 'prop-types';

const WizardErrorState = ({ title, message, onClose }) => (
  <div className="wizard-pf-complete blank-slate-pf">
    <div className="wizard-pf-success-icon">
      <span className="pficon pficon-error-circle-o" />
    </div>
    <h3 className="blank-slate-pf-main-action">{title}</h3>
    <p className="blank-slate-pf-secondary-action">{message}</p>
    {onClose ? (
      <button type="button" className="btn btn-lg btn-primary" onClick={onClose}>
        {__('Close')}
      </button>
    ) : null}
  </div>
);

WizardErrorState.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string,
  onClose: PropTypes.func
};

WizardErrorState.defaultProps = {
  message: ''
};

export default WizardErrorState;
