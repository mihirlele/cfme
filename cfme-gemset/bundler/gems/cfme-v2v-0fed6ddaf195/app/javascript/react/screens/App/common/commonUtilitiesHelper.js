const commonUtilitiesHelper = {
  getMostRecentEntityByCreationDate: entities =>
    entities && entities.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current)),

  groupBy: (items, key) =>
    items &&
    items.reduce(
      (result, item) => ({
        ...result,
        [item[key]]: [...(result[item[key]] || []), item]
      }),
      {}
    ),

  flatten: (arrayItems, flattenedArray) =>
    arrayItems && arrayItems.map(task => task.map(arrayElement => flattenedArray.push(arrayElement)))
};

export default commonUtilitiesHelper;
