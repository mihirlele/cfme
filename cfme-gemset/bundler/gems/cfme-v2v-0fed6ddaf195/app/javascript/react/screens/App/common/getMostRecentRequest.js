const getMostRecentRequest = requests =>
  requests.length && requests.reduce((prev, current) => (prev.created_on > current.created_on ? prev : current));

export default getMostRecentRequest;
