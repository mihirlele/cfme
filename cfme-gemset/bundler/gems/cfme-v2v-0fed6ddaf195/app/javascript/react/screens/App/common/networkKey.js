/**
 * returns a unique network based on current UI model
 */
export const networkKey = network => `${network.clusterId}\\${network.uid_ems || network.ems_ref}`;
