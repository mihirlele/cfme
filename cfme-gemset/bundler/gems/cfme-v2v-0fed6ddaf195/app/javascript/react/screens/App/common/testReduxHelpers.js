import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import configureMockStore from 'redux-mock-store';

const middlewares = [thunk, promiseMiddleware()];

export const generateStore = (reducers, initialState) =>
  createStore(combineReducers(reducers), initialState, applyMiddleware(...middlewares));

export const mockStore = initialState => configureMockStore(middlewares)(initialState);
