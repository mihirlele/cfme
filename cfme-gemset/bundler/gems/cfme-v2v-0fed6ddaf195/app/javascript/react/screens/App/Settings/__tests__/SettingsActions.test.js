import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import * as actions from '../SettingsActions';
import {
  fetchServersData,
  servers,
  settingsFormValues,
  fetchSettingsData,
  patchSettingsData
} from '../settings.fixures';
import { requestConversionHostsData } from '../../Mappings/screens/MappingWizard/components/MappingWizardGeneralStep/mappingWizardGeneralStep.fixtures';
import { initialState } from '../SettingsReducer';
import { mockRequest, mockReset } from '../../../../../common/mockRequests';
import { SHOW_V2V_CONVERSION_HOST_WIZARD, HIDE_V2V_CONVERSION_HOST_WIZARD } from '../SettingsConstants';

const middlewares = [thunk, promiseMiddleware()];
const mockStore = configureMockStore(middlewares);
const store = mockStore(initialState);

afterEach(() => {
  store.clearActions();
  mockReset();
});

describe('settings actions', () => {
  it('should fetch servers and return PENDING and FULFILLED action', () => {
    const { method, fetchServersUrl, response } = fetchServersData;
    mockRequest({
      method,
      url: fetchServersUrl,
      status: 200,
      response
    });
    return store.dispatch(actions.fetchServersAction(fetchServersUrl)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('should fetch servers and return PENDING and REJECTED action', () => {
    const { method, fetchServersUrl } = fetchServersData;
    mockRequest({
      method,
      url: fetchServersUrl,
      status: 500
    });
    return store.dispatch(actions.fetchServersAction(fetchServersUrl)).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('should fetch settings and return PENDING and FULFILLED action', () => {
    const { method, fetchSettingsUrl, response } = fetchSettingsData;
    mockRequest({
      method,
      url: fetchSettingsUrl,
      status: 200,
      response
    });
    return store.dispatch(actions.fetchSettingsAction(fetchSettingsUrl)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('should fetch settings and return PENDING and REJECTED action', () => {
    const { method, fetchSettingsUrl } = fetchSettingsData;
    mockRequest({
      method,
      url: fetchSettingsUrl,
      status: 500
    });
    return store.dispatch(actions.fetchSettingsAction(fetchSettingsUrl)).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('should patch settings and return PENDING and FULFILLED actions', () => {
    const { method, response } = patchSettingsData;
    servers.resources.forEach(server => {
      mockRequest({
        method,
        url: `${server.href}/settings`,
        status: 200,
        response
      });
    });
    return store.dispatch(actions.patchSettingsAction(servers.resources, settingsFormValues)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('should patch settings and return PENDING and REJECTED action', () => {
    const { method } = patchSettingsData;
    servers.resources.forEach(server => {
      mockRequest({
        method,
        url: `${server.href}/settings`,
        status: 500
      });
    });
    return store.dispatch(actions.patchSettingsAction(servers.resources, settingsFormValues)).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('should fetch conversion hosts and return PENDING and FULFILLED action', () => {
    const { fetchConversionHostsUrl } = requestConversionHostsData;
    mockRequest({
      url: fetchConversionHostsUrl,
      status: 200
    });
    return store.dispatch(actions.fetchConversionHostsAction(fetchConversionHostsUrl)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('should fetch conversion hosts and return PENDING and REJECTED action', () => {
    const { fetchConversionHostsUrl } = requestConversionHostsData;
    mockRequest({
      url: fetchConversionHostsUrl,
      status: 500
    });
    return store.dispatch(actions.fetchConversionHostsAction(fetchConversionHostsUrl)).catch(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('should dispatch show wizard action', () => {
    store.dispatch(actions.showConversionHostWizardAction());
    expect(store.getActions()).toEqual([{ type: SHOW_V2V_CONVERSION_HOST_WIZARD }]);
  });

  it('should dispatch hide wizard action', () => {
    store.dispatch(actions.hideConversionHostWizardAction());
    expect(store.getActions()).toEqual([{ type: HIDE_V2V_CONVERSION_HOST_WIZARD }]);
  });
});
