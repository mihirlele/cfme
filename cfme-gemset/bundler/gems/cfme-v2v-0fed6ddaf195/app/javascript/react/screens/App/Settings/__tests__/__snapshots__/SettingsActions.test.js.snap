// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings actions should fetch conversion hosts and return PENDING and FULFILLED action 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_CONVERSION_HOSTS_PENDING",
  },
  Object {
    "type": "FETCH_V2V_CONVERSION_HOSTS_FULFILLED",
  },
]
`;

exports[`settings actions should fetch conversion hosts and return PENDING and REJECTED action 1`] = `
Array [
  Object {
    "type": "FETCH_V2V_CONVERSION_HOSTS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "FETCH_V2V_CONVERSION_HOSTS_REJECTED",
  },
]
`;

exports[`settings actions should fetch servers and return PENDING and FULFILLED action 1`] = `
Array [
  Object {
    "type": "V2V_FETCH_SERVERS_PENDING",
  },
  Object {
    "payload": Object {
      "data": Object {
        "count": 2,
        "name": "servers",
        "pages": 1,
        "resources": Array [
          Object {
            "href": "http://0.0.0.0:8080/api/servers/42000000000001",
          },
          Object {
            "href": "http://0.0.0.0:8080/api/servers/42000000000002",
          },
        ],
        "subcount": 2,
      },
    },
    "type": "V2V_FETCH_SERVERS_FULFILLED",
  },
]
`;

exports[`settings actions should fetch servers and return PENDING and REJECTED action 1`] = `
Array [
  Object {
    "type": "V2V_FETCH_SERVERS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "V2V_FETCH_SERVERS_REJECTED",
  },
]
`;

exports[`settings actions should fetch settings and return PENDING and FULFILLED action 1`] = `
Array [
  Object {
    "type": "V2V_FETCH_SETTINGS_PENDING",
  },
  Object {
    "payload": Object {
      "data": Object {
        "otherSettings": Object {
          "mock": "data",
        },
        "transformation": Object {
          "limits": Object {
            "max_concurrent_tasks_per_ems": 10,
            "max_concurrent_tasks_per_host": 10,
          },
        },
      },
    },
    "type": "V2V_FETCH_SETTINGS_FULFILLED",
  },
]
`;

exports[`settings actions should fetch settings and return PENDING and REJECTED action 1`] = `
Array [
  Object {
    "type": "V2V_FETCH_SETTINGS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "V2V_FETCH_SETTINGS_REJECTED",
  },
]
`;

exports[`settings actions should patch settings and return PENDING and FULFILLED actions 1`] = `
Array [
  Object {
    "type": "V2V_PATCH_SETTINGS_PENDING",
  },
  Object {
    "payload": Object {
      "data": Object {
        "otherSettings": Object {
          "mock": "data",
        },
        "transformation": Object {
          "limits": Object {
            "max_concurrent_tasks_per_ems": 10,
            "max_concurrent_tasks_per_host": 10,
          },
        },
      },
    },
    "type": "V2V_PATCH_SETTINGS_FULFILLED",
  },
]
`;

exports[`settings actions should patch settings and return PENDING and REJECTED action 1`] = `
Array [
  Object {
    "type": "V2V_PATCH_SETTINGS_PENDING",
  },
  Object {
    "error": true,
    "payload": [Error: <mocked error>],
    "type": "V2V_PATCH_SETTINGS_REJECTED",
  },
]
`;
